docs/root/configuration/best_practices/edge.rst

```diff
@@ -1,5 +1,91 @@
+:ref:`listener buffer limits <envoy_api_field_Listener.per_connection_buffer_limit_bytes>` to 32 KiB
+and :ref:`cluster buffer limits <envoy_api_field_Cluster.per_connection_buffer_limit_bytes>` to 32 KiB.
+
+HTTP proxies should additionally configure:
+:ref:`use_remote_address <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.use_remote_address>` to true
+(to avoid consuming HTTP headers from external clients, see :ref:`HTTP header sanitizing <config_http_conn_man_header_sanitizing>` for details),
+and HTTP/2 settings (e.g.
+:ref:`HTTP/2 maximum concurrent streams limit <envoy_api_field_core.Http2ProtocolOptions.max_concurrent_streams>` to 100,
+:ref:`HTTP/2 initial stream window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_stream_window_size>` to 64 KiB,
+and :ref:`HTTP/2 initial connection window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_connection_window_size>` to 1 MiB).
+
+The following is a YAML example of the above recommendation.
+
+.. code-block:: yaml
+
+  overload_manager:
+    refresh_interval: 0.25s
+    resource_monitors:
+    - name: "envoy.resource_monitors.fixed_heap"
+      config:
+        # TODO: Tune for your system.
+        max_heap_size_bytes: 2147483648 # 2 GiB
+    actions:
+    - name: "envoy.overload_actions.shrink_heap"
+      triggers:
+      - name: "envoy.resource_monitors.fixed_heap"
+        threshold:
+          value: 0.95
+    - name: "envoy.overload_actions.stop_accepting_requests"
+      triggers:
+      - name: "envoy.resource_monitors.fixed_heap"
+        threshold:
+          value: 0.98
+
+  static_resources:
+    listeners:
+    - address:
+        socket_address: { address: 127.0.0.1, port_value: 443 }
+      listener_filters:
+      - name: "envoy.listener.tls_inspector"
+        typed_config: {}
+      per_connection_buffer_limit_bytes: 32768 # 32 KiB
+      filter_chains:
+      - filter_chain_match:
+          server_names: ["example.com", "www.example.com"]
+        tls_context:
+          common_tls_context:
+            tls_certificates:
+            - certificate_chain: { filename: "example_com_cert.pem" }
+              private_key: { filename: "example_com_key.pem" }
+        filters:
+        - name: envoy.http_connection_manager
+          typed_config:
+            "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+            stat_prefix: ingress_http
+            use_remote_address: true
+            common_http_protocol_options:
+              idle_timeout: 840s
+            http2_protocol_options:
+              max_concurrent_streams: 100
+              initial_stream_window_size: 65536 # 64 KiB
+              initial_connection_window_size: 1048576 # 1 MiB
+            route_config:
+              virtual_hosts:
+              - name: default
+                domains: "*"
+                routes:
+                - match: { prefix: "/" }
```

Comment:
Please add a comment here discussing timeouts. The default with this configuration will be a 15s request timeout. It would be good to have a comment discussing that, or just be explicit about the timeout and put it in the config. It would also be nice to briefly discuss the stream idle timeout for use with streaming APIs.

Reply:
Is this just for general education or is there a security angle here? I ask because I think that informs how the comment should be focussed.

Reply:
There is a security angle. We need timeouts on "all the things" so that includes streams/requests. The default timeout is 15s which is worth pointing out. More importantly, when using streaming requests, many users just disable the request timeout which leaves them exposed. They need to instead configure an idle timeout. We may just want to explicitly configure a request timeout and an idle timeout in this config with some comments about streaming.

Reply:
In the context of looking at a different issue, I realize that we should also explicitly call out the default 5 minute stream idle timeout at the HCM level.

Basically IMO we should explicitly document all of the timeout params here even if they are set to the defaults, just to call attention to things that people might need to change and why.

Reply:
Added `stream_idle_timeout` and `request_timeout`.

Reply:
Can you also set the timeout on the route to 15s (the default) and then have a comment that it must be disabled if using streaming APIs?

Reply:
Added.